test-lit/test_splitter.py

```diff
@@ -0,0 +1,126 @@
+""" test_splitter.py
+
+    Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+    See https://llvm.org/LICENSE.txt for license information.
+    SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+    
+"""
+
+import argparse
+import os
+import re
+from pathlib import Path
+
+
+class TestSplitter:
+    def __init__(self):
+        self.error_tests = []
+        self.modfile_tests = []
+        self.symbol_tests = []
+        self.generic_tests = []
+        self.unsupported_tests = []
+        self.test_types = ['ERROR', 'MODFILE', 'SYMBOL', 'GENERIC']
+        root = os.getcwd()
+        while not root.endswith("f18"):
```

Comment:
I believe you could use FLANG_SRC_DIR maybe!

Reply:
I could use this, and the source is hard to differ between person to person, but at the moment, since I'm not sure I can use the other lit variables for binaries and test files, I think I'll keep it as minimalistic as possible. I have created a global though.